### ISSUE TITLE
Errors on Ubuntu Gnome 17.04

### ISSUE THREAD

**Issue by zalloy** [nE0sIghT/chrome-gnome-shell-mirror#64]
Hi there,

Since upgrading from 16.04.1 LTS to 17.04, chrome-gnome-shell keeps throwing errors, and some of the extensions I had installed before the upgrade no longer work as they should. For example, I'm now unable to access the settings screen for Caffeine, and I now have to disable the extension to lock my screen. Additionally, I'm unable to use the extensions.gnome.org website because it keeps telling me the native host has exited. 

Any suggestions as to what I can try to fix this? I also tried to uninstall/reinstall the Chrome extension, but it says it's managed by enterprise and will not allow me to make any changes to it.

Thanks in advance,

Laurie

**Comment by jbicha** [comment]
You'll need the latest version of Caffeine for its Preferences window to work with the GNOME Shell version in Ubuntu 17.04.

**Comment by nE0sIghT** [comment]
> I'm unable to use the extensions.gnome.org website because it keeps telling me the native host has exited

Hello.

1. Edit /usr/bin/chrome-gnome-shell as root user and change
```DEBUG_ENABLED = False```
to
```DEBUG_ENABLED = True```
2. Close all Chromium windows/processes
3. Open Chromium from terminal and post output here

**Comment by zalloy** [comment]
@jbicha I have Caffeine 2.8.3-3build1 installed, which is the latest version according to the output when I try `sudo apt install caffeine`

@nE0sIghT I edited the file, however I use Google Chrome rather than Chromium. Running `google-chrome` from the terminal, I get the following output:

```
[2080:2080:0523/105437.269280:ERROR:background_mode_manager_aura.cc(13)] Not implemented reached in virtual void BackgroundModeManager::EnableLaunchOnStartup(bool)
[2199:2199:0523/105441.008950:ERROR:sandbox_linux.cc(343)] InitializeSandbox() called with multiple threads in process gpu-process. 
[2080:2080:0523/105451.371755:ERROR:display_info_provider_aura.cc(31)] Not implemented reached in virtual void extensions::DisplayInfoProviderAura::UpdateDisplayUnitInfoForPlatform(const display::Display &, extensions::api::system_display::DisplayUnitInfo *)
[2080:2080:0523/105452.764334:ERROR:navigation_entry_screenshot_manager.cc(135)] Invalid entry with unique id: 95
```

**Comment by jbicha** [comment]
@zalloy If you're using apt, you need `gnome-shell-extension-caffeine` installed. I don't think you need `caffeine` installed.

**Comment by zalloy** [comment]
@jbicha I just checked that too, and it says I also have the latest version of the extension:

`gnome-shell-extension-caffeine is already the newest version (0~git20161228-1ubuntu2).`

**Comment by nE0sIghT** [comment]
@zalloy 

Which version of chrome-gnome-shell do you using?

**Comment by zalloy** [comment]
@nE0sIghT I've got 8.2.1-1ubuntu1

Is there a newer version I should be using?

**Comment by nE0sIghT** [comment]
Try to apply fix from [this comment](https://github.com/nE0sIghT/chrome-gnome-shell-mirror/issues/57#issuecomment-294512728).

**Comment by zalloy** [comment]
@nE0sIghT I tried that, and restarted Chrome, but no change.
![selection_999 476](https://cloud.githubusercontent.com/assets/20016280/26362805/20f587a0-3fad-11e7-8284-a49d4e371ac6.png)
![selection_999 475](https://cloud.githubusercontent.com/assets/20016280/26362804/20e8b1ce-3fad-11e7-9336-37307380dff1.png)

**Comment by nE0sIghT** [comment]
Try to run ```chrome-gnome-shell``` in terminal and post output here.

**Comment by zalloy** [comment]
@nE0sIghT Here's what I got:

```
Traceback (most recent call last):
  File "/usr/bin/chrome-gnome-shell", line 20, in <module>
    import requests
ImportError: No module named 'requests'
```

The odd thing about this is that I have Python 2.7, 3.5, and 3.6 installed, and I also have the requests module installed on 3.5 and 3.6, so not sure why it's having trouble locating the module.

**Comment by nE0sIghT** [comment]
@zalloy 

You are missing Python Requests module for those python version that is used by chrome-gnome-shell

**Comment by jbicha** [comment]
@zalloy Where did you get chrome-gnome-shell from? The version packaged in Ubuntu 17.04 depends on python3-requests so you shouldn't get that error message with that version.

**Comment by zalloy** [comment]
@nE0sIghT @jbicha I had chrome-gnome-shell installed prior to upgrading, and I installed from the PPA. 

According to my system, I have python3-requests already installed:
Output from `pip install requests`:
`Requirement already satisfied: requests in /usr/local/lib/python3.6/dist-packages`

Output from `sudo apt install python3-requests`
`python3-requests is already the newest version (2.10.0-2).`

Not sure if I'm missing something here or what....

**Comment by zalloy** [comment]
@nE0sIghT @jbicha  Looking into this more deeply, it looks like I have the requests module installed for Python 2.7 and for 3.6, but for 3.5 it seems to be missing. Not sure what to do here, because it keeps telling me the requirements have been satisfied for 3.6 and won't allow me to install for 3.5.

**Comment by jbicha** [comment]
@zalloy The default Python3 in Ubuntu 17.04 is 3.5. You do not need to use pip to install python3-requests and maybe trying to use pip could cause you problems here…

**Comment by zalloy** [comment]
@jbicha I'm not sure what to do here. Running `python` in terminal gets me Python 2.7.13. Running `python3` in terminal gets me Python 3.5.3. And running `python3.6` in terminal gets me Python 3.6.1. However, checking under /usr/lib/python3.5/ shows the files are there, except for the python3-requests module. 

I've also tried reinstalling Python 3.5 and reinstalling python3-requests using Synaptic, but it's still not seeing the module. Any ideas on how I can get it to install the python3-requests module to 3.5 so that chrome-gnome-shell can see it?

**Comment by jbicha** [comment]
Sorry, I don't know enough about Python in Ubuntu to give you very specific advice here. One option is to just reinstall Ubuntu. And this time, don't manually install python3.6 or pip.

**Comment by nE0sIghT** [comment]
@zalloy

Change shebang in ```/usr/bin/chrome-gnome-shell``` file to point python2 binary.
Thas way you will fix ```chrome-gnome-shell``` execution, however your main problem with wrong python installations remains.

**Comment by zalloy** [comment]
@jbicha @nE0sIghT Since I need Python3.6 for a couple of work-related projects, not installing it is not really an option. However, after doing a bit of research, I was able to figure out what at least part of the issue was on my own, and fix it.

Now, I'm not getting the "Native host has exited" from extensions.gnome.org. But, I found that unless I run `chrome-gnome-shell` from terminal, it's unable to show my installed extensions. 

`chrome-gnome-shell` is no longer throwing the error about the missing requests module, as I found a way to install it to `/usr/lib/python3`. I didn't need to change the shebang on `/usr/bin/chrome-gnome-shell` to point to python2 at all. 

And I also fixed the settings screen on Caffeine by following the instructions at https://github.com/eonpatapon/gnome-shell-extension-caffeine to install it from the repos, so that is now working as well. 

Still not quite sure why I need to have `chrome-gnome-shell` running in terminal for the extensions site to work, but I'm not too worried about it for now, as everything else seems to be working. 

Thanks for all your help!
